OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+        private static readonly float _tapDuration = 0.1f;
+        private static readonly int _minUsefulKey = 8;
+        private static readonly int _maxUsefulKey = 350;
+        private static readonly int _maxComboLength = 7;
+        private static readonly string xboxPrefix = "xbox_";
+        private readonly IModLogger _logger;
+        private readonly IModConsole _console;
+
+        internal bool ShouldIgnore(KeyCode code)
+        {
+            UpdateCombo();
+            return _lastPressed != null && Time.realtimeSinceStartup - _timeout[(int)code] < _cooldown;
+        }
+
+        public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+        {
+            _console = console;
+            _logger = logger;
+            patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisRemovePre));
+            patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisRemovePre));
+            patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisUpdatePost));
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Instance = this;
+        }
+
+        private void UpdateCombo()
+        {
+            if (Time.realtimeSinceStartup - _lastUpdate > _updateCooldown)
+            {
+                _lastUpdate = Time.realtimeSinceStartup;
+                long hash = 0;
+                var keysCount = 0;
+                var countdownTrigger = true;
+                var keys = new int[_maxComboLength];
+                for (int code = _minUsefulKey; code < _maxUsefulKey; code++)
+                {
+                    if (Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && UnityEngine.Input.GetKey((KeyCode)code))
+                    {
+                        keys[keysCount] = code;
+                        keysCount++;
+                        if (keysCount > _maxComboLength)
+                        {
+                            hash = (int)RegistrationCode.CombinationTooLong;
+                            break;
+                        }
+                        hash = hash * _maxUsefulKey + code;
+                        if (Time.realtimeSinceStartup - _timeout[code] > _cooldown)
+                        {
+                            countdownTrigger = false;
+                        }
+                    }
+                }
+                if (!_comboRegistry.ContainsKey(hash))
+                {
+                    if (_lastPressed != null)
+                    {
+                        _lastPressed.SetPressed(false);
+                    }
+                    _lastPressed = null;
+                    return;
+                }
+                IModInputCombination combination = _comboRegistry[hash];
```

Comment:
var

Reply:
done (wonder how I missed it)

Reply:
Actually, is there any quick way to find all of these without using trial addon?